src/crdt.rs

```diff
@@ -1513,6 +1516,30 @@ mod tests {
         assert!(one && two);
     }
 
+    #[test]
+    fn gossip_request_bad_addr() {
+        let me = NodeInfo::new(
+            KeyPair::new().pubkey(),
+            "127.0.0.1:1234".parse().unwrap(),
+            "127.0.0.1:1235".parse().unwrap(),
+            "127.0.0.1:1236".parse().unwrap(),
+            "127.0.0.1:1237".parse().unwrap(),
+            "127.0.0.1:1238".parse().unwrap(),
+        );
+        let mut crdt = Crdt::new(me.clone()).expect("Crdt::new");
+        let nxt1 = NodeInfo::new(
+            KeyPair::new().pubkey(),
+            "0.0.0.0:0".parse().unwrap(),
```

Comment:
My convention has been to add a comment `<--- attack!` to the line that can be used to attack the network (which one is ncp?). I think that would help here to show that the intent of this test is to defend against a malicious node registering itself as a validator. It would also make it clear that defense against that class of attacks is incomplete, and that once complete, this test can perhaps be deleted in favor of a more generalized authorization check?